### ISSUE TITLE
Question regarding to the attention mask during training

### ISSUE THREAD

**Issue by Connor323** [JetAstra/SDAR#11]
Thanks for the great work! 

I wonder could you provide more explanation to the attention mask mentioned in Figure 1 from the paper? Based on the inference code, the attention mask for SDAR is a hybrid of causal + bi-directional attention mask, but I don't understand how the attention mask is computed during the training phase as mentioned in the Figure 1 (as shown on the right side below). Thanks a lot! 

<img width="437" height="201" alt="Image" src="https://github.com/user-attachments/assets/e1d2ed84-4bb3-4115-8707-268eb78c8840" />
